### ISSUE TITLE
RoadEventSourceBuilder.WithFeature() Doesn't Add Feature

### ISSUE THREAD

**Issue by thenadz** [WSDOT/wzdx.models#1]
As built, this function returns a whole new `RoadEventSourceBuilder` rather than adding provided feature builder to `this`. End result is no features end up in the feed. Meanwhile, `_features` is never added to.

**Comment by jhudsonWA** [comment]
Thanks! We will take a look and see what we missed. This is why it's not at 1.0 yet :)

**Comment by jhudsonWA** [comment]
@rhuebner76 can you take a look please. Thanks!

**Comment by thenadz** [comment]
Above PR appears to address in local testing. Thanks!

**Comment by rhuebner76** [comment]
@thenadz this would break the currently intended immutability of the builder methods, yet if immutability gets in the way or is unneeded, we can go back to returning this but it should be for all of the builder methods.

The tests around the field device builder source appear to collect the features as expected both independently and within the feed builder.

``` CS
        public void WithArrowBoardFeatureShouldReturnExpectedResult()
        {
            const string sourceId = "wsdot";
            const string featureId = "device-001";
            var builder = new FieldDeviceSourceBuilder(sourceId);
            builder = builder.WithFeature(featureId,
                factory => factory.ArrowBoard("005", (Point)GeometryFactory.CreatePoint(-10, 10)
                ));
            
            var result = builder.Features().ToList();
            Assert.NotEmpty(result);
            var feature = result.First();

            var device = feature.Properties;
            Assert.IsType<ArrowBoard>(device);
        }
```

I am open to discussion on design changes, but before that would you please provide an example of the builder usage where is it broken.

**Comment by thenadz** [comment]
Thanks, @rhuebner76! See discussion on #3. I think these are all rooted in the same place.

**Comment by thenadz** [comment]
Confirmed fix with 0.0.6 release. Thanks!
